apps/web/src/Providers.tsx

```diff
@@ -37,15 +36,13 @@ const Providers: React.FC<PropsWithChildren<{}>> = ({ children }) => {
           <SegmentProvider>
             <QueryClientProvider client={queryClient}>
               <AuthProvider>
-                <OptInProvider>
```

Comment:
the `OptInProvider` was only called when the application started, but we want to check opt-in when the route changes during app navigation

Reply:
I was wondering why this worked for me before but I guess it was because of this `navigate()` that was present before and triggered the useEffect:
https://github.com/novuhq/novu/pull/7017/files#diff-cc17c9f31a51ec7181c4bc7af9932388e5a3a796df090c2c17517d0f72401e97L23